clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -3856,10 +3856,19 @@ CodeGenModule::isTriviallyRecursive(const FunctionDecl *FD) {
 bool CodeGenModule::shouldEmitFunction(GlobalDecl GD) {
   if (getFunctionLinkage(GD) != llvm::Function::AvailableExternallyLinkage)
     return true;
+
   const auto *F = cast<FunctionDecl>(GD.getDecl());
   if (CodeGenOpts.OptimizationLevel == 0 && !F->hasAttr<AlwaysInlineAttr>())
     return false;
 
+  // We don't import function bodies from other named module units since that
+  // behavior may break ABI compatibility of the current unit.
+  Module *M = F->getOwningModule();
+  if (M && M->isModulePurview() &&
+      getContext().getCurrentNamedModule() != M->getTopLevelModule() &&
+      !F->hasAttr<AlwaysInlineAttr>())
+    return false;
```

Comment:
```suggestion
  if (const Module *M = F->getOwningModule();
      M && M->isModulePurview() &&
      getContext().getCurrentNamedModule() != M->getTopLevelModule() &&
      !F->hasAttr<AlwaysInlineAttr>())
    return false;
```

Should work, right?

Reply:
Oh, yes. I'll address it in the next round of changes.